tests/examples/CMakeLists.txt

```diff
@@ -40,28 +41,22 @@ else()
 endif()
 
 # Add the main awesome lua libraries.
-set(LUA_PATH2_ "\
-${TOP_SOURCE_DIR}/lib/?.lua;\
-${TOP_SOURCE_DIR}/lib/?/init.lua;\
-${TOP_SOURCE_DIR}/lib/?;\
-${TOP_SOURCE_DIR}/themes/?.lua;\
-${TOP_SOURCE_DIR}/themes/?;"
-)
+set(LUA_PATH_ "\
+${TOP_SOURCE_DIR}/lib/?.lua\\;\
+${TOP_SOURCE_DIR}/lib/?/init.lua\\;\
+${TOP_SOURCE_DIR}/lib/?\\;\
+${TOP_SOURCE_DIR}/themes/?.lua\\;\
+${TOP_SOURCE_DIR}/themes/?\\;\
+${LUA_PATH_}")
 
 # Add the C API shims.
-set(LUA_PATH3_ "\
-${TOP_SOURCE_DIR}/tests/examples/shims/?.lua;\
-${TOP_SOURCE_DIR}/tests/examples/shims/?/init.lua;\
-${TOP_SOURCE_DIR}/tests/examples/shims/?;"
-)
+set(LUA_PATH_ "\
+${TOP_SOURCE_DIR}/tests/examples/shims/?.lua\\;\
+${TOP_SOURCE_DIR}/tests/examples/shims/?/init.lua\\;\
+${TOP_SOURCE_DIR}/tests/examples/shims/?\\;\
+${LUA_PATH_}")
 
-# Done in 3 variables to avoid CMake from implicitly converting into a list.
-set(ENV{LUA_PATH} "${LUA_PATH3_}${LUA_PATH2_}${LUA_PATH_}")
-
-# Unset environment variables that would get in the way of evaluating LUA_PATH.
-unset(ENV{LUA_PATH_5_1})
-unset(ENV{LUA_PATH_5_2})
-unset(ENV{LUA_PATH_5_3})
```

Comment:
I don't think we want to drop these unsets. If one of these variables is set, then `$LUA_PATH` is ignored.